csi/controller_server.go

```diff
@@ -1065,19 +1066,27 @@ func (cs *ControllerServer) cleanupSnapshot(sourceVolumeName, id string) error {
 	return nil
 }
 
-func (cs *ControllerServer) cleanupBackupVolume(sourceVolumeName, id string) error {
+func (cs *ControllerServer) cleanupBackup(sourceVolumeName, id string) error {
 	backupVolumeName, backupName := sourceVolumeName, id
-	backupVolume, err := cs.getBackupVolume(backupVolumeName)
+	backupVolumes, err := cs.getBackupVolumes(backupVolumeName)
 	if err != nil {
 		return err
 	}
-	if backupVolume != nil && backupVolume.Name != "" {
-		if _, err = cs.apiClient.BackupVolume.ActionBackupDelete(backupVolume, &longhornclient.BackupInput{
-			Name: backupName,
-		}); err != nil {
-			return err
+	for _, bv := range backupVolumes {
+		// The BackupDelete API actually doesn't care about bv when deleting a backup.
+		// The bv is there for backward compatibility.
+		// Any bv will work
+		if bv.Name != "" {
+			_, err = cs.apiClient.BackupVolume.ActionBackupDelete(bv, &longhornclient.BackupInput{
```

Comment:
Will there be an error if the bv does not contain the specified backup?

Reply:
No, please see comment https://github.com/longhorn/longhorn-manager/pull/3519#discussion_r1941987374